napari/_app_model/actions/_viewer_actions.py

```diff
@@ -0,0 +1,132 @@
+        title=CommandId.VIEWER_INC_DIMS_LEFT.title,
+        callback=_viewer_actions.increment_dims_left,
+    ),
+    Action(
+        id=CommandId.VIEWER_INC_DIMS_RIGHT,
+        title=CommandId.VIEWER_INC_DIMS_RIGHT.title,
+        callback=_viewer_actions.increment_dims_right,
+    ),
+    Action(
+        id=CommandId.VIEWER_FOCUS_AXES_UP,
+        short_title=CommandId.VIEWER_FOCUS_AXES_UP.title,
+        title=CommandId.VIEWER_FOCUS_AXES_UP.description,
+        callback=_viewer_actions.focus_axes_up,
+    ),
+    Action(
+        id=CommandId.VIEWER_FOCUS_AXES_DOWN,
+        short_title=CommandId.VIEWER_FOCUS_AXES_DOWN.title,
+        title=CommandId.VIEWER_FOCUS_AXES_DOWN.description,
+        callback=_viewer_actions.focus_axes_down,
+    ),
+    Action(
+        id=CommandId.VIEWER_ROLL_AXES,
+        short_title=CommandId.VIEWER_ROLL_AXES.title,
+        title=CommandId.VIEWER_ROLL_AXES.description,
+        callback=_viewer_actions.roll_axes,
+    ),
+    Action(
+        id=CommandId.VIEWER_TRANSPOSE_AXES,
+        short_title=CommandId.VIEWER_TRANSPOSE_AXES.title,
+        title=CommandId.VIEWER_TRANSPOSE_AXES.description,
+        callback=_viewer_actions.transpose_axes,
+    ),
+    Action(
+        id=CommandId.VIEWER_TOGGLE_SELECTED_LAYER_VISIBILITY,
+        title=CommandId.VIEWER_TOGGLE_SELECTED_LAYER_VISIBILITY.title,
+        callback=_viewer_actions.toggle_selected_layer_visibility,
+    ),
+    Action(
+        id=CommandId.VIEWER_TOGGLE_CONSOLE_VISIBILITY,
+        title=CommandId.VIEWER_TOGGLE_CONSOLE_VISIBILITY.title,
+        callback=_viewer_actions.toggle_console_visibility,
+    ),
+]
+
+
+def _ndisplay_toggle(viewer: Viewer):
+    viewer.dims.ndisplay = 2 + (viewer.dims.ndisplay == 2)
+
+
+def _get_current_ndisplay_is_3D(viewer: Viewer):
+    return viewer.dims.ndisplay == 3
+
+
+# these are separate because they include menu entries
+VIEWER_ACTIONS.extend(
+    [
+        Action(
+            id=CommandId.TOGGLE_VIEWER_NDISPLAY,
+            title=CommandId.TOGGLE_VIEWER_NDISPLAY.title,
+            menus=[
+                {'id': MenuId.MENUBAR_VIEW, 'group': '1_render', 'order': 0}
+            ],
+            callback=_ndisplay_toggle,
+            toggled=ToggleRule(get_current=_get_current_ndisplay_is_3D),
+        ),
+        ViewerToggleAction(
+            id=CommandId.VIEWER_TOGGLE_GRID,
+            title=CommandId.VIEWER_TOGGLE_GRID.title,
+            viewer_attribute="grid",
+            sub_attribute="enabled",
+            menus=[
+                {'id': MenuId.MENUBAR_VIEW, 'group': '1_render', 'order': 0},
+            ],
+        ),
+    ],
+)
+
+# TODO: importing down here because I'm not sure this is the correct place for
+# this action
+from napari._qt.menus.file_menu import show_shortcuts  # noqa
+
+VIEWER_ACTIONS.append(
+    Action(
+        id=CommandId.NAPARI_SHOW_SHORTCUTS,
+        title=CommandId.NAPARI_SHOW_SHORTCUTS.title,
+        callback=show_shortcuts,
+    )
+)
```

Comment:
Hi @kne42, I need to move `show_shortcuts` as well in  #4865, as we are deleting `napari/_qt/menus/file_menu.py`. I am also unsure of where best to put this function. I thought the `show_shorcuts` function would be classified as a qt action (?) as it involves opening a dialogue (and it currently sits inside `napari/_qt/`) but I see it doesn't use any qt functions, it does use `Viewer` and gets `window` from viewer. Any insight here? :grimacing: 

If it is indeed a qt action, maybe, it could be in a new file: `napari/_qt/_qapp_model/qactions/_viewer.py`
If it is not, then I think this new file you created would be a good place.

We could coordinate PRs once we've decided on a place for it?

cc @DragaDoncila

Reply:
I agree that it should be moved to the Qt-specific area as you suggested